msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/BiGaussian.java

```diff
@@ -104,32 +103,66 @@
    * @param direction a {@link Enum} object. This decides for which half we want to calculate X
    *        value.
    */
-  private double InterpolationX(MultiKeyMap<Integer, Triplet> horizontalSlice, int mu,
-      double halfHeight, int leftBound, int rightBound, int roundedmz, Direction direction) {
+  private double InterpolationX(List<Triplet> horizontalSlice, int mu, double halfHeight,
+      int leftBound, int rightBound, int roundedmz, Direction direction) {
 
     int i = mu;
-    double Y1 = 0;
-    double Y2 = 0;
+    Double Y1 = null;
+    Double Y2 = null;
```

Comment:
Instead of using ```null``` to say whether these ```Doubles``` have been set, I suggest keeping as ```double``` and initializing with ```Double.NaN```.